File: be/StockFlow/src/main/java/com/madebyzino/StockFlow/service/TransactionService.java
package com.madebyzino.StockFlow.service;

import com.madebyzino.StockFlow.dto.transaction.TransactionHistoryResponse;
import com.madebyzino.StockFlow.dto.transaction.TransactionSearchCondition;
import com.madebyzino.StockFlow.repository.TransactionQueryRepository;
import lombok.RequiredArgsConstructor;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.stereotype.Service;

@Service
@RequiredArgsConstructor
public class TransactionService {

    private final TransactionQueryRepository transactionQueryRepository;

    public Page<TransactionHistoryResponse> searchTransactionHistory(TransactionSearchCondition condition, Pageable pageable) {
        return transactionQueryRepository.searchTransactions(condition, pageable);
    }
}

File: fe/src/services/transactionService.ts
import api from "./api"; 
import { PageResponse } from '../types/page';
import { TransactionSearchCondition, TransactionHistoryResponse } from '../types/transaction';

export const transactionService = {
    searchTransactionHistory: async (
        condition: TransactionSearchCondition,
        page: number,
        size: number
    ): Promise<PageResponse<TransactionHistoryResponse>> => {
        const response = await api.get(`/api/transactions/history`, {
            params: {
                ...condition, 
                page, 
                size,
            },
        });
        return response.data;
    },
};